### PR TITLE
[FIX] SearchScheme warnings on 32bit and Windows

### PR DESCRIPTION
**Pull request by cpockrandt**

### CODE REVIEW

File: tests/index/test_find2_index_approx.h [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #define TESTS_FIND2_INDEX_APPROX_H_
 
 #include <mutex>
+#include <time.h>
```
Comment (h-2) [inline tests/index/test_find2_index_approx.h]: Never use C-headers directly if there are are C++ headers with namespace std (`<ctime>`).

Please have a look how random is done in c++11:
http://en.cppreference.com/w/cpp/numeric/random/rand

**Comment by h-2** [comment]
Please not that there is an additional warning caused by you commits on g++8:
```
[ 14%] Building CXX object tests/index/CMakeFiles/test_index_stree_iterators.dir/test_index_stree_iterators.cpp.o
In file included from /.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.cpp:37:
/.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.h: In instantiation of 'void SEQAN_TEST_test_find2_index_approx_small_test() [with bool speed_up_dummy_to_prevent_compilation_of_unused_tests_ = true]':
/.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.cpp:44:5:   required from here

/.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.h:478:9: error: range-based 'for' expression of type 'auto' has incomplete type

         for (auto occ : getOccurrences(iter))
         ^~~
In file included from /.../FreeBSD-11.1-RELEASE_g++8_64/include/seqan/index.h:236,
[ 14%] Linking CXX executable ../../bin/test_find2_index_approx
                 from /.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.h:40,
                 from /.../FreeBSD-11.1-RELEASE_g++8_64/tests/index/test_find2_index_approx.cpp:37:
```

**Comment by h-2** [comment]
Did your last push incorporate fixes for g++8, too?

**Comment by cpockrandt** [comment]
@h-2 Since you closed #2208, can you merge this PR?

**Pull request merged**